patterns/ecs-v1/java

```diff
@@ -4,14 +4,14 @@ JAVAFILE (?:[A-Za-z0-9_. -]+)
 #Allow special <init>, <clinit> methods
 JAVAMETHOD (?:(<(?:cl)?init>)|[a-zA-Z$_][a-zA-Z$_0-9]*)
 #Line number is optional in special cases 'Native method' or 'Unknown source'
-JAVASTACKTRACEPART %{SPACE}at %{JAVACLASS:class}\.%{JAVAMETHOD:method}\(%{JAVAFILE:file}(?::%{NUMBER:line})?\)
+JAVASTACKTRACEPART %{SPACE}at %{JAVACLASS:[log][origin][class][name]}\.%{JAVAMETHOD:[log][origin][function]}\(%{JAVAFILE:[log][origin][file][name]}(?::%{INT:[log][origin][file][line]})?\)
```

Comment:
these are all ECS existing fields except for `log.origin.class.name`
have looked into what APM does for Java - using it's own (`error`) custom structure (which makes sense since it never really sends a stack-trace frame but pretty much always a full trace in a single document).

alternatively to avoid clashing with the standard we could use `java.log.origin.class_name` or similar ...

Reply:
@webmat do you guys have any input seeing this explicitly as `log.origin.class.name`?
this might obviously be open to potential future clashes with ECS but I am not finding a better name :name_badge: 
would be better to have a common ECS field such as `log.origin.owner.name` since it's not always a class?

happy to move this to the ECS repo later, but I would like to start with smt reasonable here. 
there was a previous attempt to [introduce `log.origin.module`](https://github.com/elastic/ecs/issues/785) (understand the use-case presented there fitted `log.logger`) this is a different use-case - the name might fit here despite potential confusion with Java's module names.

Reply:
Yeah I still think the PR you reference makes sense to revisit, despite the latest discussions.

We haven't had time to look into it, so I would recommend a custom field like `java.log.origin.class_name` for now. 

In addition to having a potential conflict with a future version of ECS, adding a custom fields within existing ECS fields also causes confusion. Either users wonder why the `log.origin.class.name` field is not present in the ECS docs; or if ECS adds it later as a different name, now you'd have to populate the ECS field  (e.g. `log.origin.module.name`) with the same value.

When the fields is clearly nested in a tech-specific section of your schema, then I think users have an easier time understanding that this is not (yet) part of ECS.